apps/meteor/client/components/Omnichannel/CustomFieldsForm/index.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { ReactElement } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { CustomSelectInput } from './CustomSelectInput';
+import { CustomTextInput } from './CustomTextInput';
+import { useCustomFields } from './hooks/useCustomFields';
+
+const CustomFieldsForm = (): ReactElement | null => {
+	const { data, isLoading, isError, error } = useCustomFields();
+
+	const { register, control } = useFormContext<{ livechatData: Record<string, unknown> }>();
+
+	if (!data || isLoading) {
+		return null;
+	}
+
+	if (isError) {
+		throw error;
+	}
+
+	return (
+		<>
+			{data.customFields
+				.filter(
+					(customField) =>
+						// TODO: REMOVE FILTER ONCE THE ENDPOINT SUPPORTS A SCOPE PARAMETER
+						customField.scope === 'room',
+				)
+				.map((customField) => {
+					switch (customField.type) {
+						case 'input':
+							return (
+								<CustomTextInput
+									key={customField._id}
+									data={customField}
+									{...register(`livechatData.${customField._id}`, {
+										validate: {
+											required: (value) => !!value,
```

Comment:
`required`doesnt need to be within `validade`, validate is usually for custom validations. 
`rules={{ required: true, [...] }}` should work.